File: wallets/bitcoind/regtest/src/main/java/bisq/wallets/regtest/Os.java
package bisq.wallets.regtest;

import lombok.Getter;

import java.util.Locale;

public enum Os {
    LINUX("linux"),
    MAC_OS("macos"),
    WINDOWS("win");

    @Getter
    private final String canonicalName;

    Os(String canonicalName) {
        this.canonicalName = canonicalName;
    }

    public static boolean isLinux() {
        return isLinux(getOsName());
    }

    public static boolean isLinux(String osName) {
        return osName.contains("linux");
    }

    public static boolean isMacOs() {
        return isMacOs(getOsName());
    }

    public static boolean isMacOs(String osName) {
        return osName.contains("mac") || osName.contains("darwin");
    }

    public static boolean isWindows() {
        return isWindows(getOsName());
    }

    public static boolean isWindows(String osName) {
        return osName.contains("win");
    }

    public static String getOsName() {
        return System.getProperty("os.name").toLowerCase(Locale.US);
    }

    public static String getOsVersion() {
        return System.getProperty("os.version");
    }
}

File: build-logic/gradle-tasks/src/main/kotlin/bisq/gradle/common/OS.kt
package bisq.gradle.common

import java.util.*

enum class OS(val osName: String) {
    LINUX("linux"),
    MAC_OS("macos"),
    WINDOWS("win")
}

fun getOS(): OS {
    val osName = getOSName()
    if (isLinux(osName)) {
        return OS.LINUX
    } else if (isMacOs(osName)) {
        return OS.MAC_OS
    } else if (isWindows(osName)) {
        return OS.WINDOWS
    }

    throw IllegalStateException("Running on unsupported OS: $osName")
}

private fun isLinux(osName: String): Boolean {
    return osName.contains("linux")
}

private fun isMacOs(osName: String): Boolean {
    return osName.contains("mac") || osName.contains("darwin")
}

private fun isWindows(osName: String): Boolean {
    return osName.contains("win")
}

fun getOSName(): String {
    return System.getProperty("os.name").toLowerCase()
}
